Content.Shared/Body/Systems/SharedBodySystem.Body.cs

```diff
@@ -283,6 +283,15 @@ public IEnumerable<BodyPartSlot> GetBodyAllSlots(EntityUid bodyId, BodyComponent
         SoundSpecifier? gibSoundOverride = null
         )
     {
+        if (deleteItems)
+        {
+            var items = _inventory.GetHandOrInventoryEntities(bodyId);
+            foreach (var item in items)
+            {
+                QueueDel(item);
+            }
+        }
+
```

Comment:
theres already a further check over the inventory/held items later at line 326, shouldnt it be handled there? (deleting instead of attaching to grid)

Reply:
IMO deleting items should not be functionality on gibBody. It should be done before the gib call is made. Item deletion in the gibBody function just adds additional complexity and argument bloat when it should be handled in a separate function.